tests/executables_src/testUserInputValidator.cc

```diff
@@ -620,19 +615,13 @@ BOOST_AUTO_TEST_CASE(testDeepBackwardsPropagation) {
   // test two consecutive values, only the first being discarded at the lowest level and the second is accepted
   upstrIn.setAndWrite(12);
   upstrIn.setAndWrite(3);
-  test.stepApplication();
-  BOOST_TEST(midstreamIn.readNonBlocking());
-  BOOST_TEST(midstreamIn == 13); // first value is coming from upstream
-  BOOST_TEST(downstrIn.readNonBlocking());
-  BOOST_TEST(downstrIn == 14); // first value is coming from upstream/midstream
 
+  test.stepApplication();
   BOOST_TEST(downstrIn.readLatest()); // just observe final state, because intermediate states might be subject
   BOOST_TEST(downstrIn == 5);         // to race conditions
   BOOST_TEST(midstreamIn.readLatest());
   BOOST_TEST(midstreamIn == 4);
-  BOOST_TEST(upstrIn.readLatest());
```

Comment:
Why is this line removed?

Reply:
It failed because there is no update. But I should have replaced it with !readLatest

Reply:
I ahve added the test with !readLatest